File: 50pow(x, n)/pow(x, n).js
/**
 * @param {number} x
 * @param {number} n
 * @return {number}
 */
var myPow = function(x, n) {
    let result = 1
    const sign = Math.sign(n)
    while (n)
    {
        if (n & 1) result *= x
        x *= x
        n = Math.trunc(n / 2)
    }
    return Object.is(sign, 1) ? result : 1 / result
};

File: 50pow(x, n)/pow(x, n).py
import numpy

class Solution:
    def myPow(self, x: float, n: int) -> float:
        result = 1
        sign = numpy.sign(n)
        while n:
            if n & 1: result *= x
            x *= x
            n = int(n / 2)
        return result if sign == 1 else 1 / result 
